Login  page created

AnshumaliSuri opened the pull request:
### **About**

1. Info about the PR: This is my PR to propose  login  and verification application screens.
2. Name: Anshumali Suri
3. GitHub ID:[ Anshumali Suri](https://github.com/AnshumaliSuri)

### _Approach_

1. The package I used was the default flutter package  flutter/material.dart
2. I used ListView widget for basic structure of both my screens.
3. Used boolean variables to set state of buttons and checklist.

### _Screenshots_



<img width="476" alt="Screenshot 2022-12-20 at 10 38 36 PM" src="https://user-images.githubusercontent.com/114004856/208726180-bf30f6fe-2fb5-4711-8ee1-e59c5af85f2a.png">

<img width="476" alt="Screenshot 2022-12-20 at 10 40 05 PM" src="https://user-images.githubusercontent.com/114004856/208726272-69c9799c-0c16-47e9-9bc5-0d16863be7c1.png">


<img width="476" alt="Screenshot 2022-12-20 at 10 39 14 PM" src="https://user-images.githubusercontent.com/114004856/208726340-15a3503c-96ca-4189-9c3f-1c4fdd4e03e5.png">

AnshumaliSuri committed:
Login  page created

RitwikSingh28 commented on prac2-pages/anshumali_suri/lib/main.dart:
```diff
@@ -0,0 +1,202 @@
+}
+
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  TextEditingController nameController = TextEditingController();
```
A better way to initialize the TextEditingControllers is through overriding the initState() method, and then you must free the resources through overriding the dispose() method. 

https://api.flutter.dev/flutter/widgets/TextEditingController-class.html
**Please refer to the above documentation for dispose() method**

AnshumaliSuri replied:
Resolved in commit [Update main.dart](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/7d63d65c799f15f3d81c7dde0ead33dff3ae0ba8)

RitwikSingh28 commented on prac2-pages/anshumali_suri/lib/main.dart:
```diff
@@ -0,0 +1,202 @@
+        child: ListView(
+          children: <Widget>[
+
+            Container(
```
Try to minimize the use of Container in your Flutter projects, they only serve to increasing complexity if you are not going to specify any other parameters. 
Moreover, since you have not specified the dimensions of the Container (i.e. its height and width), alignment won't work.

Try to change your code such that you make minimal use of Containers while also not losing any of the functionality and features.

AnshumaliSuri replied:
Resolved in commit  [Reduced usage of Container widget](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/480ac135a3b666197b5d3f6400c14416f023cb06)

RitwikSingh28 commented on prac2-pages/anshumali_suri/lib/main.dart:
```diff
@@ -0,0 +1,202 @@
+
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  TextEditingController nameController = TextEditingController();
+  TextEditingController passwordController = TextEditingController();
```
It is also a good practice to keep your instance variables as private. :thinking: 
This leads to a better implementation of the OOP principles.

AnshumaliSuri replied:
Resolved in commit [Update main.dart](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/7d63d65c799f15f3d81c7dde0ead33dff3ae0ba8)

RitwikSingh28 commented on prac2-pages/anshumali_suri/lib/main.dart:
```diff
@@ -0,0 +1,202 @@
+
+                )),
+            Container(
+              padding: const EdgeInsets.all(10),
```
If you just wish to add padding in a container, use the Padding widget instead. 
Leads to a much more optimized code and improves the performance of your app.

AnshumaliSuri replied:
Resolved in commit [Changed Container widget to Padding](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/4b8f5ce14d50ba439d243a747e119661daa05135)

RitwikSingh28 commented on prac2-pages/anshumali_suri/lib/main.dart:
```diff
@@ -0,0 +1,202 @@
+                  ),
+                ),
+                value: value,
+                onChanged: (value) => setState(() => this.value = value!),
```
value! tells the compiler that it is null-safe. I believe you wanted to use the logical NOT, so please look into this as well..

AnshumaliSuri replied:
I deliberately used null safety operator and did not wanted to use not operator.
<img width="846" alt="Screenshot 2022-12-21 at 1 12 30 PM" src="https://user-images.githubusercontent.com/114004856/208848719-30ddf84f-5091-4a55-8d07-20bd69835d32.png">

RitwikSingh28 reviewed (changes requested):
The UI looks magnificent.... Great work @AnshumaliSuri ...
Before we can merge your PR, there are some changes that we would like you to make.
Please complete these changes within the next **2 days**.

AnshumaliSuri committed:
Changed Container widget to Padding

AnshumaliSuri committed:
Reduced usage of Container widget

AnshumaliSuri committed:
Update main.dart

Used TextEditingController class with. initState() method and dispose() method.Also made instance variables private by using "_" to name them.

RitwikSingh28 reviewed (changes requested):
Amazing work @AnshumaliSuri :sparkles: !!
Here's another change that I want you to make, can you try to center the entire main content onto the center of the screen? As it looks, the content is all shifted up, it would look better if it were to be in the center of the page.
Don't you agree as well? :thinking: 
Try to make the aforementioned change

AnshumaliSuri committed:
Centered whole page by adding padding on top.

AnshumaliSuri commented:
> Amazing work @AnshumaliSuri ✨ !!
> Here's another change that I want you to make, can you try to center the entire main content onto the center of the screen? As it looks, the content is all shifted up, it would look better if it were to be in the center of the page.
> Don't you agree as well? 🤔
> Try to make the aforementioned change

Resolved in the commit [Centered whole page by adding padding on top.](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/eda5bf10ae14a1b1b5a8f99a7ef9b725f8f17d9f)

RitwikSingh28 reviewed (changes requested):
All issues are resolved. :sparkles: 
Try to add Navigation now between the two pages, so that when you click on Login, you get directed to the second page.

AnshumaliSuri committed:
Added functionality of login button

AnshumaliSuri committed:
Added functionality of "return to login" button.

AnshumaliSuri commented:
> All issues are resolved. ✨
> Try to add Navigation now between the two pages, so that when you click on Login, you get directed to the second page.

Navigation added successfully in commits [Added functionality of login button](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/e84a9dfd070317b2336562bd5c850978419c507b) and   [Added functionality of "return to login" button.](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/b295945dadd361a108930614f0286c909e7026a5)

RitwikSingh28 reviewed (changes requested):
Magnificent work so far @AnshumaliSuri !! :sparkles: 

I request you to make one more modification.
**Details**:
> As it is, this page is not adaptive to different screen sizes. Try to make it adaptive utilizing the ScreenUtil package in Flutter.

Once done, this PR would be considered complete and we will merge it into the main branch!
Thank you!! :smile:

AnshumaliSuri committed:
Made the 1st screen adaptive

AnshumaliSuri committed:
Made 2nd page adaptive

AnshumaliSuri commented:
> Magnificent work so far @AnshumaliSuri !! ✨
> 
> I request you to make one more modification. **Details**:
> 
> > As it is, this page is not adaptive to different screen sizes. Try to make it adaptive utilizing the ScreenUtil package in Flutter.
> 
> Once done, this PR would be considered complete and we will merge it into the main branch! Thank you!! 😄

@RitwikSingh28 Completed the task successfully in commits [Made the 1st screen adaptive](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/a3e4b7941607ec3bc368ff139c7bb1fd581aaacc) and [Made 2nd page adaptive](https://github.com/RitwikSingh28/GitHubPrac/pull/15/commits/04edee33a2681e4bcac75ebc93539c8fec65d86f).

RitwikSingh28 reviewed (approved):
Congratulations :partying_face: !

There are no issues left, this is an excellent PR. Great work once again..
Merging into main..
